Doc/library/stdtypes.rst

```diff
@@ -4106,8 +4106,10 @@ its contents cannot be altered after it is created; it can therefore be used as
 a dictionary key or as an element of another set.
 
 Non-empty sets (not frozensets) can be created by placing a comma-separated list
-of elements within braces, for example: ``{'jack', 'sjoerd'}``, in addition to the
-:class:`set` constructor.
+of elements within braces, for example: ``{'jack', 'sjoerd'}``.  Non-empty sets can
+be created by using a setcomprehension, for example ``{x for x in iterable}``.
+Also, set comprehensions can make both empty and non-empty sets.  In addition to
+the :class:`set` constructor.
```

Comment:
There is no need to say "Non-empty sets (not frozensets) can be created by using a set comprehension..." and then add "Also, set comprehensions can make both empty and non-empty sets". Just remove the "non-empty from the first sentence and delete the whole second sentence.